### PR TITLE
Filter Min = 0 Fix

### PR DESCRIPTION
**Pull request by simon-leech**
# Filter Min Fix

## Description
I have added a new filter_minmax query that returns an array of both min and max values. This removed some of the complexity in the generateMinMax function as you no longer need to pass in a string of "min" or "max".

I have added a new filter.minmax array that gets set to [filter.min, filter.max] initially. The benefit of this is that by extracting the query response from the filter.min and filter.max values, we separate providing a specific min or max in the configuration vs getting them from the database.

I have resolved the issue with clear and reload of the filter, it will now update the filter values using the query.
I have added a new message that displays when the min and max are identical.
image

## GitHub Issue

 - [#1301](https://github.com/GEOLYTIX/xyz/issues/1301)

## Type of Change
Please delete options that are not relevant, and select all options that apply. 

- ✅ Bug fix (non-breaking change which fixes an issue)

## How have you tested this? 
Ran this using a field that contained 0 as the minimum field value. 

## Testing Checklist 

- ✅ Existing Tests still pass
- ✅ Ran locally on my machine

## Code Quality Checklist

- ✅ My code follows the guidelines of XYZ
- ✅ My code has been commented
- ✅ Documentation has been updated
- ✅ New and existing unit tests pass locally with my changes
- ✅ Main has been merged into this PR

### CODE REVIEW

**Review by dbauszus-glx — CHANGES_REQUESTED** [review]
I really struggle to wrap my head around this. How can I test this.

But I also have issues with the integer filter.

If the layer has only one value and min max are the same then the filter should be disabled.

No input is possible.

image

The min max should be run again if I close and open the filter. I added a value to the example and had to reload the locale to see the filter changing.

```js
"field": "minutes",
"filter": {
  "type": "integer"
},
```

**Comment by simon-leech** [comment]
I have added a new `filter_minmax` query that returns an array of both min and max values. This removed some of the complexity in the `generateMinMax` function as you no longer need to pass in a string of "min" or "max". 

I have added a new `filter.minmax` array that gets set to `[filter.min, filter.max] initially. The benefit of this is that by extracting the query response from the `filter.min` and `filter.max` values, we separate providing a specific min or max in the configuration vs getting them from the database. 

I have resolved the issue with clear and reload of the filter, it will now update the filter values using the query. 
I have added a new message that displays when the min and max are identical. 
image

File: lib/ui/layers/filters.mjs [hunk 1]
```diff
@@ -133,25 +134,20 @@ function filter_null(layer, filter) {
...
     locale: layer.mapview.locale.key,
     layer: layer.key,
     table: layer.tableCurrent(),
     field: filter.field,
   })}`);
 
-  // If the response contains response.min or response.max - we can assume the query was successful.
-  // Have to check that response is not null as well
-  response ??= {}
-  if (response[minmax]) {
```
Comment (RobAndrewHurst) [inline lib/ui/layers/filters.mjs:151]: Nice one removing this! 

We must still take note that `if(0)` is falsy.

**Review by dbauszus-glx — APPROVED** [review]
It should be possible to define the step in config.

Added to docs.

```js
  // if (!filter.step) {

  //   filter.step = filter.type === 'integer' ? 1 : 0.01;
  // }

  filter.step ??= filter.type === 'integer' ? 1 : 0.01;
```

I've updated the docs for the generateMinMax to include field, min, max [optional]

Implicit check can be done by checking min, max not a number, ie undefined.

Promise<void>, ie no return doesn't require a return in description.

```js
/**
@function generateMinMax
@description
Queries the min and max bounds for a filter field. The min and max bounds can be implicit in the filter config.

@param {Object} layer MAPP layer object.
@param {Object} filter Filter object.
@param {string} filter.field Field to filter.
@param {numeric} [filter.min] Min bounds.
@param {numeric} [filter.max] Max bounds.
*/

async function generateMinMax(layer, filter) {

  const response = await mapp.utils.xhr(`${layer.mapview.host}/api/query?${mapp.utils.paramString({
    template: `field_minmax`,
    locale: layer.mapview.locale.key,
    layer: layer.key,
    table: layer.tableCurrent(),
    field: filter.field,
  })}`);

  // Assign min, max from response if not a number.
  const min = isNaN(filter.min) ? response.minmax[0]: filter.min
  const max = isNaN(filter.max) ? response.minmax[1]: filter.max

  // Parse min, max as interger or float depending on type.
  // Round integer to be correct up or down.
  filter.minmax[0] = filter.type === 'integer'? Math.round(min) : parseFloat(min)
  filter.minmax[1] = filter.type === 'integer'? Math.round(max) : parseFloat(max)
}
```

File: lib/ui/layers/filters.mjs (hunk 1 above)
Comment (dbauszus-glx) [reply lib/ui/layers/filters.mjs:151]: This should be resolved.

**Pull request merged**